code/modules/bitrunning/objects/disks.dm

```diff
@@ -30,13 +34,33 @@
 	. += span_info("It has been used to select: <b>[choice_made]</b>.")
 	. += span_notice("It cannot make another selection.")
 
+/// Handles loading our stuff onto avatars
+/obj/item/bitrunning_disk/proc/load_onto_avatar(mob/living/carbon/human/neo, mob/living/carbon/human/avatar, external_load_flags)
+	return NONE
+
+
 /obj/item/bitrunning_disk/ability
 	desc = "A disk containing source code. It can be used to preload abilities into the virtual domain. Duplicate abilities will be ignored."
 	/// The selected ability that this grants
 	var/datum/action/granted_action
 	/// The list of actions that this can grant
 	var/list/datum/action/selectable_actions = list()
 
+/obj/item/bitrunning_disk/ability/load_onto_avatar(mob/living/carbon/human/neo, mob/living/carbon/human/avatar, external_load_flags)
+	if(external_load_flags & DOMAIN_FORBIDS_ABILITIES)
+		return LOAD_BLOCKED
+
+	if(isnull(granted_action))
+		return LOAD_FAILED
+
+	var/datum/action/our_action = new granted_action()
+
+	if(locate(our_action.type) in avatar.actions)
```

Comment:
why create it first at all if all we need is the type (IE granted_action)

Reply:
funnily enough this is old bitrunning code I just Didn't Think About when porting it that sounds insane now in hindsight

Reply:
addressed in both places